src/main/java/com/mongodb/jdbc/MongoSQLResultSet.java

```diff
@@ -132,136 +114,65 @@ protected Object getObject(BsonValue o, int columnType) throws SQLException {
             return null;
         }
         switch (columnType) {
-            case Types.ARRAY:
-                // not supported
-                break;
             case Types.BIGINT:
+            case Types.INTEGER:
+            case Types.SMALLINT:
+            case Types.TINYINT:
                 return getInt(o);
             case Types.BINARY:
-                return o.asBinary();
+                return o.asBinary().getData();
             case Types.BIT:
-                return getBoolean(o);
-            case Types.BLOB:
-                // not supported
-                break;
             case Types.BOOLEAN:
                 return getBoolean(o);
             case Types.CHAR:
```

Comment:
Looks you missed moving CHAR with LONGVARCHAR, VARCHAR, etc....

Reply:
done